src/transformers/tokenization_utils_fast.py

```diff
@@ -831,32 +831,29 @@ def train_new_from_iterator(
-            if "special_tokens" in post_processor:
-                for key in post_processor["special_tokens"]:
-                    tokens = post_processor["special_tokens"][key]["tokens"]
-                    if special_tokens_map is not None:
-                        tokens = [special_tokens_map.get(token, token) for token in tokens]
-                    post_processor["special_tokens"][key]["tokens"] = tokens
-                    for token in tokens:
+            _post_processors = [post_processor]
+            # If the post-processor is of type Sequence, handle the individual processors
+            if "processors" in post_processor:
+                _post_processors.extend(post_processor["processors"])
+
+            for _post_processor in _post_processors:
+                if "special_tokens" in _post_processor:
+                    for key in _post_processor["special_tokens"]:
+                        tokens = _post_processor["special_tokens"][key]["tokens"]
+                        if special_tokens_map is not None:
+                            tokens = [special_tokens_map.get(token, token) for token in tokens]
+                        _post_processor["special_tokens"][key]["tokens"] = tokens
+                        _post_processor["special_tokens"][key]["ids"] = [
+                            tokenizer.token_to_id(token) for token in tokens
+                        ]
+
+                for special_token in ["cls", "sep"]:
+                    if special_token in _post_processor:
+                        token, _ = _post_processor[special_token]
+                        if special_tokens_map is not None and token in special_tokens_map:
+                            token = special_tokens_map[token]
                         token_id = tokenizer.token_to_id(token)
-                        if token_id is None:
-                            raise ValueError(
-                                "Attempted to set a token in the post processor that does not exist in the mapping"
-                            )
-
-                    post_processor["special_tokens"][key]["ids"] = [tokenizer.token_to_id(token) for token in tokens]
-
-            for special_token in ["cls", "sep"]:
-                if special_token in post_processor:
-                    token, _ = post_processor[special_token]
-                    if special_tokens_map is not None and token in special_tokens_map:
-                        token = special_tokens_map[token]
-                    token_id = tokenizer.token_to_id(token)
-                    if token_id is None:
-                        raise ValueError(
-                            "Attempted to set a token in the post processor that does not exist in the mapping"
-                        )
-                    post_processor[special_token] = [token, token_id]
+                        _post_processor[special_token] = [token, token_id]
 
             trained_tokenizer_json["post_processor"] = post_processor
```

Comment:
Thanks for working on this!! 👍 
Should this also be updated to set to `_post_processors`?

Also, would be great to add a test for this so that we can catch this in the future! Let me know if you would like to contribute the test or I can add it! 🤗

Reply:
Thanks for the feedback! The reason for using `_post_processor` on the final block is to ensure that if Roberta or BERT processors are ever part of the Sequential postprocessor, their `cls `and `sep` token IDs will be correctly updated.  I can revert this if you think it’s unnecessary.

Including a unit test is a great idea - I’ll start working on it.

Reply:
Added a test case for the Sequence post-processor. @itazap, let me know what you think. Should anything else be added or changed?